### DISCUSSION TITLE
Serverside code being included in _app bundle

### DISCUSSION THREAD

**Discussion by georgebutter** [vercel/next.js#49264] [Help] [answered]
### Summary

I have an external library that contains my components and the Graphql queries that are required to render those components. The external library exports uncompiled TS which is imported and compiled by SWC. The GraphQL query is only called in `getStaticProps` and yet it is still included in the client side bundle. Here is an example:

```
import {MyComponent} from "@library/MyComponent";
import {getComponentStaticProps, Props} from "@library/MyComponent/getStaticProps";

const Page: React.FC<Props> = (props) => <MyComponent {...props} />;

export async function getStaticProps(): Promise<GetStaticPropsResult<Props>> {
  const props = await getComponentStaticProps();
  return {
    props,
  };
}
```

Is this the expected behaviour or have I structured things in a way that stops SWC from purging the query from the app bundle?

**Comment by icyJoseph** [comment, marked as answer]
I think, and according to the next-code elimination tool, that this should not happen:

```jsx
// This is the code that is bundled for the client-side:

import { MyComponent } from '@library/MyComponent';
const Page = (props) => <MyComponent {...props} />;

```

Perhaps you are importing that library into the _app bundle by "accident"? 

For example, if you have, this module, which you intent to only use server side:

```js
export const ENDPOINT_URL = "something"
```

But, you import `ENDPOINT_URL` into a page, in order to show the URL in the UI, then tree shaking might fail to only include that, under some circumstances.

For example here: https://github.com/icyJoseph/next-js-react-gbg/commit/3610a9dc32ba29f202c3b31bd44e7f67cb334ee0, I had imported the pokemon url, from a module that also used crypto, and browserify packaged a bunch of supporting code, that was unused, just for that. Do not blindly rely on tree-shaking.

**Reply by georgebutter** [reply]
Really appreciate the clarification that this is not the expected behaviour and is avoidable. Now I just need to figure out how to avoid it in my use case. Cheers. 👍
